ocagent.go

```diff
@@ -148,14 +150,17 @@ func (ae *Exporter) Start() error {
 	var err = errAlreadyStarted
 	ae.startOnce.Do(func() {
 		ae.mu.Lock()
-		defer ae.mu.Unlock()
-
 		ae.started = true
 		ae.disconnectedCh = make(chan bool, 1)
 		ae.stopCh = make(chan bool)
 		ae.backgroundConnectionDoneCh = make(chan bool)
+		ae.mu.Unlock()
 
-		ae.setStateDisconnected()
+		if err := ae.connect(); err == nil {
+			ae.setStateConnected()
+		} else {
+			ae.setStateDisconnected(err)
+		}
 		go ae.indefiniteBackgroundConnection()
```

Comment:
Not sure if we still need this?

Reply:
We still need to start the `indefiniteBackgroundConnection()` to ensure that if the connection is lost the dial happens again.

Reply:
Yes we do need that indefiniteBackground connection.